### PR TITLE
Document the data_path option and Hass.io Mosquitto add-on compatability

### PR DESCRIPTION
**Pull request by kasparsd**
- Fix the spelling of the `data_path` option.
- Mention that `data_path` has to be relative to the Home Assistant's shared directory and can't be any other system directory.
- Document the `mqtt` configuration and the specific user/password requirements for the [official Mosquitto add-on](https://github.com/home-assistant/hassio-addons/tree/master/mosquitto#known-issues-and-limitations).
- Specify the configuration menu location for each option.
- Add more spacing for readability for a few sections.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,24 +22,33 @@ Version 1.5.1 contains breaking changes and requires re-formating of the add-on
...
+Configure the add-on via the Hass.io front-end under "Supervisor (Hass.io) → Dashboard → zigbee2mqtt".
+
+The configuration closely mirrors that of [zigbee2mqtt itself](https://www.zigbee2mqtt.io/information/configuration.html), with a couple of key differences:
+
 1. Hass.io requires add-on configuration in JSON format, rather than YAML. If you don't understand the difference, you can use a YAML-to-JSON converter.
-2. An additional top-level `data-path` option is required. Set this to the path where you would like the add-on to persist data. Defaults to `/share/zigbee2mqtt`. Note that both `config` and `share` directories are mapped into the container (read-write) and are available to you.
+
+2. An additional top-level `data_path` option is required which defaults to `/share/zigbee2mqtt`. This is the path where the add-on should persist the data. The path must be relative to the Home Assistant shared data directory (which is `/usr/share/hassio` for Hass.io). Note that both `config` and `share` directories are mapped into the container (read-write) and are available to you.
```
Comment (kasparsd) [inline README.md]: This is the core change -- we're changing `data-path` to `data_path` and adding a the note about the file path being relative to the HA's shared directory.

As a first time user I tried setting it to my existing zigbee2mqtt configuration location `/home/pi/zigbee2mqtt` which obviously didn't work.

File: README.md [hunk 2]
```diff
@@ -22,24 +22,38 @@ Version 1.5.1 contains breaking changes and requires re-formating of the add-on
...
+
+### MQTT settings
+
+Depending on your configuration, the MQTT server config may need to include the port, typically `1883` or `8883` for SSL communications. For example, `mqtt://core-mosquitto:1883` for [Hass.io's Mosquitto addon](https://github.com/home-assistant/hassio-addons/blob/master/mosquitto/README.md).
 
-- Depending on your configuration, the MQTT server config may need to include the port, typically `1883` or `8883` for SSL communications. For example, `mqtt://core-mosquitto:1883` for Hass.io's Mosquitto addon.
-- To find out which serial ports you have exposed go to **Supervisor (used to be Hass.io) > System > Host system > Hardware**
+Ensure the user credentials specified under the `mqtt` section (`user` and `password`) are correct and have write access to the MQTT server. Additional [configuration is required](https://github.com/home-assistant/hassio-addons/tree/master/mosquitto#known-issues-and-limitations) when the `anonymous` option is enabled in the [Hass.io's Mosquitto addon](https://github.com/home-assistant/hassio-addons/blob/master/mosquitto/README.md).
```
Comment (kasparsd) [inline README.md]: Here we mention the additional configuration required for the official Mosquitto server add-on compatibility.

**Comment by kasparsd** [comment]
This is ready for code review!

Thank you so much for creating and maintaining this amazing add-on!

**Comment by danielwelch** [comment]
Love it, thanks!

**Pull request merged**